### DISCUSSION TITLE
ForAllOtherMembers removed

### DISCUSSION THREAD

**Discussion by jeparrish** [LuckyPennySoftware/AutoMapper#3833] [NA]
With the latest version, ForAllOtherMembers was removed.  I understand the usage of the "MemberList.None", but I am doing conditions in ForAllOtherMembers and only mapping if the value is not null.  How would I accomplish this now without have to map every single member?

`.ForMember(dest => dest.PhoneNumbers, opt => opt.Ignore())
.ForAllOtherMembers(opts => 
                        opts.Condition((src, dest, srcMember) => srcMember != null)
)`

**Comment by lbargaoanu** [comment]
You can use `ForAllMembers` and check the member. Or you can implement `ForAllOtherMembers` in your own code. Whatever you need in AM for that, we can expose somehow.
That being said, this kind of mapping is not that useful. Because when someone adds a new member, it's difficult to understand what's going on. It won't end well.

**Reply by JakubHolovsky** [reply]
@lbargaoanu is there any elegant workaround to basically replace:

.ForAllOtherMembers(o => o.Ignore())

I had a look at multiple solutions but they either not compile anymore or do not work.

I tried ForAllMembers Ignore and then explicitly map certain properties but they are still ignored.

**Reply by lbargaoanu** [reply]
We only want to support what the upgrade guide says, `MemberList.None`.

**Reply by JakubHolovsky** [reply]
@lbargaoanu I get that but that doesn't supplement the functionality of `.ForAllOtherMembers(o => o.Ignore())` so my question still stands, is there any elegant workaround? Or any suggestions?

**Reply by lbargaoanu** [reply]
Yes, don't use such an abomination :) Or maybe use a mapper that is designed to work that way. AM is not.

**Reply by JakubHolovsky** [reply]
OK, maybe I am gonna put here an example and you might tell me if I am missing something:

```
public class MyDto
    {
        public Guid Id { get; set; }

        public string A { get; set; }

        public string B { get; set; }
    }

    public class MyA
    {
        public Guid Id { get; set; }

        public string A { get; set; }
    }

    public class MyB
    {
        public Guid Id { get; set; }

        public string B { get; set; }
    }
```

CreateMap<MyA, MyDto>();
CreateMap<MyB, MyDto>(); // SKIP mapping "Id" here but map "B".

This is a super simple example which could be easily solved by adding ForMember Id Ignore() but let's say that there is multiple properties that match this way and I want to skip mapping them without adding too many ignores. From my testing "MemberList.None" doesn't do anything good in here. Maybe there is something I missed and this can be easily achieved?

**Reply by lbargaoanu** [reply]
You're going against AM's very purpose. But if you must, there is nothing stopping you from writing an `Ignore` function that takes a bunch of property names.

**Reply by JakubHolovsky** [reply]
Got it. However still relies on me providing a property list. My aim here is to Map only what I defined and nothing else. From your know how. Can I get the mapped properties in a list so I can just check what I have mapping for? So that way I can simply ignore the ones that are not in my mapped list? 

I saw bunch of examples but they no longer work.

I guess I am kind of going in circles and wanting to have something like "IgnoreWithoutExplicitMapping" which I believe was once in AM but no longer is.

**Reply by lbargaoanu** [reply]
We are not going to help you shoot yourself in the foot. But good luck with that :)

**Reply by jbogard** [reply]
Yes, this is a mapping situation we don't want to support. AutoMapper is about enforcing conventions, you're saying "don't have conventions", which is the antithesis of what AutoMapper is designed for. So we're being very judicious about supporting features that align with the design philosophy.
